src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -447,14 +603,10 @@ public override bool MoveNext()
                         state = 2;
                         goto case 2;
                     case 2:
-                        while (_enumerator.MoveNext())
+                        if (_enumerator.MoveNext())
```

Comment:
Should `SelectListIterator.ToArray` use `List<T>.Enumerator` as well, for the same reason?